src/jira/api.rs

```diff
@@ -126,12 +128,36 @@ impl Session for JiraSession {
     }
 
     fn comment_issue(&self, key: &str, comment: &str) -> Result<()> {
+        #[derive(Serialize)]
+        struct VisibilityReq {
+            r#type: String,
+            value: String,
+        }
+        
         #[derive(Serialize)]
         struct CommentReq {
             body: String,
+            visibility: Option<VisibilityReq>,
         }
 
-        let req = CommentReq { body: comment.to_string() };
+        let req = match &self.restrict_comment_visibility_to_role {
```

Comment:
if you wanted to simplify this a bit, you could do something like:

```rust
let mut req = CommentReq { 
    body: comment.to_string(),
    visibility: None
}
if let Some(role) = self.restrict_comment_visibility_to_role {
    req.visibility = Some(VisibilityReq { /*...*/ })
}
```

or another idea:
```rust
let mut req = CommentReq { 
    body: comment.to_string(),
    visibility: self.restrict_comment_visibility_to_role.map(|r| VisibilityReq { /* ... */})
}
```